### PR TITLE
Use runCtx instead of of config.Opts for suggesting suggestions. 

### PR DESCRIPTION
**Pull request by tejal29**
Relates to #5485

This is a refactor, to break change in smaller PR for #5485.
When adding deploy error codes #4933, i missed the part where `kubectx` gets called directly. We have that information in `runCtx`.  It easier to use the evaluated context there instead of re-writing logic to look up `kubeContext`

Pros
1) Remove dependency on `pkg/skaffold/config` and `pkg/skaffold/kubernetes/context` from errors. 
2) In upcoming PR, we need the logic to determine minikube vs remote cluster for "Internal System Error". Using runCtx is much easier than again evaluating kubecontext in `deploy/errors` package.

### CODE REVIEW

File: pkg/skaffold/errors/errors.go [hunk 1]
```diff
@@ -36,16 +37,16 @@ const (
...
-
-	// Report issue text
-	reportIssueText = "If above error is unexpected, please open an issue https://github.com/GoogleContainerTools/skaffold/issues/new to report this error"
 )
 
 var (
-	setOptionsOnce sync.Once
-	skaffoldOpts   config.SkaffoldOptions
+	// Report issue text
+	reportIssueText = fmt.Sprintf("If above error is unexpected, please open an issue %s to report this error", constants.GithubIssueLink)
```
Comment (MarlonGamez) [inline pkg/skaffold/errors/errors.go:44]: should we just move this to const as well?

**Review by MarlonGamez — COMMENTED** [review]
refactor looks good, have a small question

File: pkg/skaffold/errors/errors.go (hunk 1 above)
Comment (tejal29) [reply pkg/skaffold/errors/errors.go:44]: I usually prefer `fmt.Sprintf` over concatenation but i can change it.

Comment (MarlonGamez) [reply pkg/skaffold/errors/errors.go]: oh my bad, I didn't realize you can't use it to create const values. Up to you if you want to leave it or not

**Review by MarlonGamez — APPROVED** [review]
Looks good to me :)

**Pull request merged**